### ISSUE TITLE
Improve SQLite queries in the external integration modules

### ISSUE THREAD

**Issue by noise-kngdm** [wazuh/wazuh#10871]
|Component|
|---|
| External integration modules |

## Description  
In recent developments we came across some problems related to the SQLite queries used by the external integration modules.  

- There are some SQLite queries in the AWS module that aren't correct, for instance, the following query returns not the last key **but the first one**:
https://github.com/wazuh/wazuh/blob/f381fc2883555b4a996c4ac6548de7c545c96618/wodles/aws/aws_s3.py#L1818-L1829  

- Some queries don't consider the prefix when asking for a key, so they might return unwanted values, a behavior that could end generating duplicate alerts. That's because since the dates used to filter the logs sent by AWS are parsed using the log key returned by those queries, you could run a query that returns a log key pertaining to a set of files with a prefix different than the one specified. If the date in that log key is older than the one that was actually needed, the module will process duplicate logs.  

The goal of this issue is to investigate and solve those problems in both the Google Cloud and AWS modules, since those are the modules that use SQLite databases.

## Checks
- [x] Check AWS queries
- [x] Check GCP queries
- [x] Test changes
- [x] Test that it's working in an agent 
- [x] Changelog (`CHANGELOG.md`)

**Comment by noise-kngdm** [comment]
## Issue update
We have reviewed every SQLite query in both the AWS and GCP modules. The GCP module queries had not any error, but we found some mistakes in the AWS ones. For example, the following one isn't considering the `suffix` of the logs stored.  

https://github.com/wazuh/wazuh/blob/4b28038a6e4342b9bdcc13a1cdee5acba6f5e7df/wodles/aws/aws_s3.py#L437-L449  

This could be troublesome because the dates used to filter the logs sent by AWS are parsed using the log key returned by queries like this, and this incorrect query could return a value pertaining another set of logs, which could potentially end in receiving duplicate alerts.

**Comment by noise-kngdm** [comment]
## Issue update
When testing the changes made to the code to check that everything was working fine, we decided to compare the results with the ones that the master branch would return with the troublesome environment. We realized that, even though the prefix and the suffix weren't considered comparing with the `log_key`, they were stored in the `bucket_path` field.  

Therefore, when executing the module looking for logs with a suffix, only the requested logs will be considered, like in the following example execution: 

```
root@6fa78efae015:/var/ossec/wodles/aws# wpy ./aws_s3.py -d2 -b wazuh-aws-wodle-cloudtrail -t cloudtrail -s 2021-nov-01  -L test_suffix
DEBUG: +++ Debug mode on - Level: 2
DEBUG: +++ Working on XXXXXXXXXXXX - us-west-1
DEBUG: +++ Marker: AWSLogs/test_suffix/XXXXXXXXXXXX/CloudTrail/us-west-1/2021/11/01
DEBUG: ++ Skipping previously processed file: AWSLogs/test_suffix/XXXXXXXXXXXX/CloudTrail/us-west-1/2021/11/01/XXXXXXXXXXXX_CloudTrail_us-west-1_20211101T0000Z_rUQtGNUqmrPZMCnu.json.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/test_suffix/XXXXXXXXXXXX/CloudTrail/us-west-1/2021/11/02/XXXXXXXXXXXX_CloudTrail_us-west-1_20211102T0000Z_qJRrDloAWNoOMRAy.json.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/test_suffix/XXXXXXXXXXXX/CloudTrail/us-west-1/2021/11/03/XXXXXXXXXXXX_CloudTrail_us-west-1_20211103T0000Z_OtoPYJqXyVMnScng.json.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/test_suffix/XXXXXXXXXXXX/CloudTrail/us-west-1/2021/11/08/XXXXXXXXXXXX_CloudTrail_us-west-1_20211108T0000Z_rUQtGNUqmrPZMCnu.json.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/test_suffix/XXXXXXXXXXXX/CloudTrail/us-west-1/2021/11/11/XXXXXXXXXXXX_CloudTrail_us-west-1_20211111T0000Z_rUQtGNUqmrPZMCnu.json.gz
DEBUG: ++ Skipping previously processed file: AWSLogs/test_suffix/XXXXXXXXXXXX/CloudTrail/us-west-1/2021/11/15/XXXXXXXXXXXX_CloudTrail_us-west-1_20211115T0000Z_rUQtGNUqmrPZMCnu.json.gz
DEBUG: ++ Found new log: AWSLogs/test_suffix/XXXXXXXXXXXX/CloudTrail/us-west-1/2021/11/22/XXXXXXXXXXXX_CloudTrail_us-west-1_20211122T0000Z_rUQtGNUqmrPZMCnu.json.gz
DEBUG: +++ DB Maintenance
DEBUG: +++ DB Maintenance
```

## Conclussions
All the wrong queries where fixed in #9993 and no more changes will be needed.

**Comment by CarlosRS9** [comment]
To be fixed in #9993
